src/blockdata/block.rs

```diff
@@ -332,6 +337,12 @@ impl Block {
         let push = input.script_sig.instructions_minimal().next().ok_or(Bip34Error::NotPresent)?;
         match push.map_err(|_| Bip34Error::NotPresent)? {
             script::Instruction::PushBytes(b) if b.len() <= 8 => {
+                // Check that the number is encoded in the minimal way.
+                // As the number is encoded in little-endian, the last
+                // byte is the most significant one.
+                if b.last() == Some(&0x00) {
+                    return Err(Bip34Error::NonMinimalPush(b.to_vec()));
+                }
```

Comment:
Is this correct? Can't a number start with `0080` to force it to be positive if it'd otherwise be negative?

Reply:
Good catch! This is incorrect.

Reply:
Fixed this in a much cleaner way